backend/internal/flow/flowexec/engine.go

```diff
@@ -526,3 +566,235 @@ func createExecutionAttempt(nodeRecord *common.NodeExecutionRecord, nodeResp *co
 
 	return attempt
 }
+
+// publishNodeExecutionStartedEvent publishes an observability event when node execution starts.
+func publishNodeExecutionStartedEvent(
+	ctx *EngineContext,
+	node core.NodeInterface,
+	obsSvc observability.ObservabilityServiceInterface,
+) {
+	if obsSvc == nil || !obsSvc.IsEnabled() {
+		return
+	}
+
+	// Get node execution record to determine step number and attempt
+	record := ctx.ExecutionHistory[node.GetID()]
+	stepNumber := len(ctx.ExecutionHistory) + 1
+	attemptNumber := 1
+	if record != nil {
+		stepNumber = record.Step
+		attemptNumber = len(record.Executions) + 1
+	}
+
+	evt := event.NewEvent(
+		ctx.FlowID, // Use FlowID as TraceID
+		string(event.EventTypeFlowNodeExecutionStarted),
+		event.ComponentFlowEngine,
+	).
+		WithStatus(event.StatusInProgress).
+		WithData(event.DataKey.FlowID, ctx.FlowID).
+		WithData(event.DataKey.FlowType, string(ctx.FlowType)).
+		WithData(event.DataKey.NodeID, node.GetID()).
+		WithData(event.DataKey.NodeType, string(node.GetType())).
+		WithData(event.DataKey.StepNumber, fmt.Sprintf("%d", stepNumber)).
+		WithData(event.DataKey.AttemptNumber, fmt.Sprintf("%d", attemptNumber)).
+		WithData(event.DataKey.AppID, ctx.AppID)
+
+	// Set executor details if applicable (only for executor-backed nodes)
+	if node.GetType() == common.NodeTypeTaskExecution {
+		if executableNode, ok := node.(core.ExecutorBackedNodeInterface); ok {
+			executor := executableNode.GetExecutor()
+			if executor != nil && record != nil {
+				record.ExecutorName = executor.GetName()
+				record.ExecutorType = executor.GetType()
```

Comment:
Have we used these variables?

Reply:
We are not using these variables. These are just string representation. The real value will be showing in the event. Introduce these keys to avoid duplicate string representations